tests/Fixer/Phpdoc/PhpdocVarWithoutNameFixerTest.php

```diff
@@ -172,21 +295,156 @@ public function provideFixVarCases()
+            'testSingleLineNoProperty' => [
+                <<<'EOF'
+<?php
+
+/** @var Foo\Bar $bar */
+$bar;
+EOF
+            ],
+            'testMultiLineNoProperty' => [
                 <<<'EOF'
 <?php
+
+/**
+ * @var Foo\Bar $bar
+ */
+$bar;
+EOF
+            ],
+            'testVeryNestedInlineDoc' => [
+                <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * @var array {
+     *     @var array $secondLevelOne   {
+     *         {@internal This should not break}
+     *         @var int $thirdLevel
+     *     }
+     *     @var array $secondLevelTwo   {
+     *         @var array $thirdLevel     {
+     *             @var string $fourthLevel
+     *         }
+     *         @var int   $moreThirdLevel
+     *     }
+     *     @var int   $secondLevelThree
+     * }
+     */
+    public $nestedFoo;
+}
+EOF
+                ,
+                <<<'EOF'
+<?php
+
+class Foo
+{
     /**
-     * @param int[] $stuff {
-     *     @var int $foo
+     * @var array $nestedFoo {
+     *     @var array $secondLevelOne   {
+     *         {@internal This should not break}
+     *         @var int $thirdLevel
+     *     }
+     *     @var array $secondLevelTwo   {
+     *         @var array $thirdLevel     {
+     *             @var string $fourthLevel
+     *         }
+     *         @var int   $moreThirdLevel
+     *     }
+     *     @var int   $secondLevelThree
      * }
+     */
+    public $nestedFoo;
+}
+EOF
+            ],
+        ];
+    }
+
+    /**
+     * @requires PHP 7.0
+     * @dataProvider provideFixVar70Cases
      *
-     * @return void
+     * @param string $expected
+     * @param string $input
      */
+    public function testFixVar70($expected, $input)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixVar70Cases()
+    {
+        return [
+            'anonymousClass' => [
+                <<<'EOF'
+<?php
+
+class Anon
+{
+    public function getNewAnon()
+    {
+        return new class()
+        {
+            /**
+             * @var string
+             */
+            public $stringVar;
 
+            public function getNewAnon()
+            {
+                return new class()
+                {
+                    /**
+                     * @var string
+                     */
+                    public $stringVar;
+                };
+            }
+        };
+    }
+}
+EOF
+                ,
+                <<<'EOF'
+<?php
+
+class Anon
+{
+    public function getNewAnon()
+    {
+        return new class()
+        {
+            /**
+             * @var $stringVar string
+             */
+            public $stringVar;
+
+            public function getNewAnon()
+            {
+                return new class()
+                {
+                    /**
+                     * @var $stringVar string
+                     */
+                    public $stringVar;
+                };
+            }
+        };
+    }
+}
```

Comment:
@dmvdbrugge Could you add a test case with multiple properties such as e.g.:
```php
class Foo {
    /**
     * @var int $foo
     */
    private $foo, $bar;
}
```

In such case, the variable name should not be removed from the `@var` tag as it would become ambiguous.

Also [PSR-19](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc-tags.md#examples-15) shows a sample with variation of this scenario but I'm not sure we should explicitly support it for now as PSR-19 is still a draft.

```php
class Foo {
    private
        /**
         * @var int
         */
        $foo,
        /**
         * @var int
         */
        $bar;
}
```

Reply:
The scope of this issue was to only fix classy properties. I think that the request made here is out of scope and can be resolved with a new issue and PR